Doc/library/multiprocessing.shared_memory.rst

```diff
@@ -69,6 +69,7 @@ copying of data.
       memory block may be larger or equal to the size requested.
       When attaching to an existing shared memory block,
       the *size* parameter is ignored.
+      On macOS, the maximum size of a shared memory block is 140256418463744 bytes.
```

Comment:
At the very least please add some commas, but it might be more helpful to use a saner and more human readable unit:
```suggestion
      On macOS, the maximum size of a shared memory block is 127 TiB (140,256,418,463,744 bytes).
```

Reply:
Also, it would be useful to add a reference to where this number comes from. I can't see anything on the linked issue mentioning it.

Reply:
Hi! Thank you for your comment! This number has been computed this way: https://gist.github.com/nilleb/e60e7adbfe2afaf9d727231c59b01100

I have tested it on macOS Sequoia 15.6.1.
That is the best I can afford for the moment, I don't have access to older operating systems.

Reply:
Ok, if this is an experimentally found number we should remove it, there are no guarantees.

Instead, say something like 'the maximum size is around 120 TiB'.

However, I'm now less sure about including this warning in the documentation, given the limit doesn't come from a documented or well-known property of macOS, and could change without warning. cc @python/macos-team for thoughts.

Reply:
Searching further, I have found this SO post https://stackoverflow.com/a/77078609

Where they mention 0x00007ffffe000000 that corresponds to 140737454800896 (so a bit more that the size I get).
This would be compatible with the fact that the memory for a python process is limited to 128TiB on a 64bits macOS OS (and hence the maximum allowed value depends on the memory already allocated by python itself).

I would propose to add a link to https://github.com/apple-oss-distributions/xnu/blob/xnu-8796.141.3/osfmk/mach/arm/vm_param.h#L137

And a paraphrase of the explanation above.

WDYT?

Reply:
OK, here is a further commit for your review! (the math role seems to be available, thank you!)